controllers/handlers_utils.go

```diff
@@ -130,12 +131,12 @@ func deployContentOfSecret(ctx context.Context, deployingToMgmtCluster bool, des
 	}
 
 	return deployContent(ctx, deployingToMgmtCluster, destConfig, destClient, secret, data,
-		clusterSummary, mgmtResources, logger)
+		clusterSummary, mgmtResources, []string{}, logger)
```

Comment:
We need to fetch annotation sub resources here as well. All referenced Kind should be treated equally